src/compiler/pre-transform/codemods/component-meta-to-define-meta.test.ts

```diff
@@ -0,0 +1,57 @@
+import pkg from '@storybook/addon-svelte-csf/package.json' with { type: 'json' };
+import type { Comment, Component } from 'svelte/compiler';
+import { print } from 'svelte-ast-print';
+import { describe, it } from 'vitest';
+
+import { transformComponentMetaToDefineMeta } from './component-meta-to-define-meta';
+
+import { parseAndExtractSvelteNode } from '#tests/extractor';
+
+describe(transformComponentMetaToDefineMeta.name, () => {
```

Comment:
1. Done in fdebf5ca6486d5e08faef98fede1176156a1be2e
2. This one is tough.
   I'm not sure what should be expected outcome. We move this variable declaration to module tag?
   Currently it is transforming to:

   ```ts
   const { Story } = defineMeta({
      	component: WithParameters,
      	parameters: { ...parameters, baz: 'yes' }
   });
   ```
   which of course would produce an incorrect behaviour at runtime.
   
   This case might need more work. Because apart of `parameters` prop, there could be other properties _(from meta)_ being referenced in variables, e.g. `tags`, `play`, etc.

Reply:
I'm thinking either:

1. we attempt to move it to the module script. This can be tricky, because moving it could require moving _other_ variables as well that it references. This is very edgy though, so maybe not something we need to support.
2. or we detect the reference and log a warning to the user that the legacy API-usage couldn't be supported. Ignore the file completely, tell them to migrate the file to the new API.

Reply:
1. This is too much complex work. I can see it when I just finished initial support of moving from instance tag to module:
   - addon import declaration _(and transforming it)_
   - optional stories component import
   - export const meta _(and transforming it)_
   
   I'd have to walk on `ObjectExpression` on `export const meta` or `<Meta>` attributes _(props)_, and finding identifiers to each meta entry at the root level _(depth = 0)_. Now, imagine some very edge cases with depth level > 1 entries _(e.g. for `parameters`)_ located in the instance tag. 😅
   
2. I need to think on how to display this warning - as in how to detect references.